File: examples/sequence.cpp
#include "board.h"
#include <unistd.h>
#include <iostream>

using namespace std;

int main(){
    useconds_t threeMic = 3000000;

    Board board(112, "/dev/i2c-1");
    cout << "Arming board" << endl;
    board.arm();
    usleep(threeMic);
    cout << "Setting all motors to full throttle" << endl;
    board.setSpeedAll(255);
    usleep(threeMic);
    cout << "Stopping motor 2" << endl;
    board.stopM2();
    usleep(threeMic);
    cout << "Stopping motor 1" << endl;
    board.stopM1();
    usleep(threeMic);
    cout << "Setting motor 1 to half throttle" << endl;
    board.setSpeedM1(-128);
    usleep(threeMic);
    cout << "Reversing M1 direction" << endl;
    board.reverseM1();
    usleep(threeMic);
    cout << "Stopping all motors" << endl;
    board.stopAll();
    cout << "Disarming board" << endl;
    board.disarm();
    usleep(threeMic);
    cout << "Activating relay" << endl;
    board.buck.enableInput();
    usleep(threeMic);
    cout << "Deactivating relay" << endl;
    board.disarm();
    
    return 0;
}

File: examples/sequence.py
from pypowerboard import Board
import time

board = Board(112, "/dev/i2c-1")
# time.sleep(1)
# board.arm()
# time.sleep(1)
# board.set_speed_m1(255)
# board.set_speed_m2(-128)
# time.sleep(3)
# board.stop_all()
# time.sleep(1)
# board.disarm()
# time.sleep(2)


# print(board.buck.enable_regulator())
buck = board.buck()
buck.enable_regulator()
# print(buck)
# buck.enable_regulator()
